adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,967 @@
+
+'''
+
+.[apititle]#Default constructor#
+[source,role=synopsis,id=api:requirements-ctor]
+----
+template <typename... Requirements>
+requirements(Requirements... values);
+----
+
+_Constraints_:
+* [code]#is_requirement_v# returns [code]#true# for each type in
+  [code]#Requirements#; and
+* [code]#Requirements# contains at most one
+  [code]#kernel_bundle<bundle_state::executable>#.
+
+_Effects_: Constructs a [code]#requirements# object representing the set of
+requirements specified via the [code]#values# parameter pack.
+
+_Remarks_: Unless otherwise specified, if an instance of a requirement appears
+more than once in the [code]#values# parameter pack, the [code]#requirements#
+object behaves as if it had only been specified once.
+
+'''
+
+.[apititle]#requirements::register-accessors#
+[source,role=synopsis,id=api:register-accessors]
+----
+void register-accessors(handler& h) const; // exposition only
+----
+
+_Effects_: Calls [code]#h.require# for each [code]#accessor# passed to the
+constructor of this [code]#requirements# object.
+
+'''
+
+.[apititle]#requirements::register-events#
+[source,role=synopsis,id=api:register-events]
+----
+void register-events(handler& h) const; // exposition only
+----
+
+_Effects_: Calls [code]#h.depends_on# for each [code]#event# or
+[code]#std::vector<event># passed to the constructor of this
+[code]#requirements# object.
+
+'''
+
+.[apititle]#requirements::register-kernel-bundle#
+[source,role=synopsis,id=api:register-kernel-bundle]
+----
+void register-kernel-bundle(handler& h) const; // exposition only
+----
+
+_Effects_: Calls [code]#h.use_kernel_bundle# if a [code]#kernel_bundle# in
+executable state was passed to the constructor of this [code]#requirements#
+object and has no effect otherwise.
+
+'''
+
+.[apititle]#requirements::has-tracking#
+[source,role=synopsis,id=api:has-tracking]
+----
+bool has-tracking() const; // exposition only
+----
+
+_Returns_: [code]#true# if this [code]#requirements# object was constructed with
+a [code]#tracking# object with tracking enabled, and [code]#false# otherwise.
+
+'''
+
+.[apititle]#tracking# constructor
+[source,role=synopsis,id=api:tracking-ctor]
+----
+namespace sycl::khr {
+
+tracking(bool enabled=true);
+
+}
+----
+
+_Effects_: Construct a [code]#tracking# object, representing a requirement that
+a command must be submitted such that its state can be tracked via an
+[code]#event# when [code]#enabled# is [code]#true#.
+
+{note}If an [code]#event# is _not_ required, [code]#tracking(false)# should be
+expected to introduce a small amount of overhead compared to providing no
+[code]#tracking# requirement.{endnote}
+
+== New free functions
+
+=== Kernel launch
+
+// Launch a basic parallel_for with a function object.
+// New form of queue::parallel_for(range, ...)
+.[apititle]#launch# (kernel function)
+[source,role=synopsis,id=api:launch]
+----
+namespace sycl::khr {
+
+template <typename KernelType, typename... Requirements>
```

Comment:
Why do we have separate overloads for each dimensionality rather than having a template parameter?

Reply:
This is a simplification designed to make the specification easier to read.

In SYCL 2020, these functions are all defined as being templated on `int Dims`, but there's a special note in [Section 4.9.4.2.2](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_parallel_for_invoke) saying that `parallel_for` must be invocable with braced initializer lists.  A compiler can't infer the `Dims` parameter from a braced initializer list, so all implementations today have to define overloads for each dimension.

We decided that for this extension we'd just expose the separate overloads.